homeassistant/helpers/script.py

```diff
@@ -134,7 +134,7 @@ def __init__(
         self,
         hass: HomeAssistant,
         script: "Script",
-        variables: Optional[Sequence],
+        variables: TemplateVarsType,
```

Comment:
This should still have `Optional` ?

Reply:
Already defined as such, https://github.com/home-assistant/core/blob/dbcd5f4c2ce7694016ffba6bd8db2c6a8f256e3a/homeassistant/helpers/typing.py#L16

Reply:
Isn't that weird to have optional be part of the "main" type. I feel like optional should be context specific.

Reply:
Gut feeling says yes, but in helpers.template where the variables by that type end up being used it is truly always optional. If there are some places where passing some variables is actually a must (can't tell offhand), then it would be good to remove optionality from the type itself.